lua/telescope/pickers.lua

```diff
@@ -626,6 +628,24 @@ function Picker:find()
       local start_time = vim.loop.hrtime()
 
       local prompt = self:_get_next_filtered_prompt()
+      if self.allow_location_input == true then
+        local filename, line_number, column_number = utils.separate_file_path_location(prompt)
+
+        if line_number or column_number then
+          state.set_global_key("prompt_location", { row = line_number, col = column_number })
+          self:refresh_previewer()
+        elseif state.get_global_key "prompt_location" then
+          state.set_global_key("prompt_location", nil)
+          self:refresh_previewer()
+        end
+
+        -- it is important to continue behaving as if there is no location in prompt
+        prompt = filename
+      elseif state.get_global_key "prompt_location" then
+        -- in case new picker that does not support locations is opened clear the location
+        -- without refreshing previewer
+        state.set_global_key("prompt_location", nil)
+      end
```

Comment:
this doesnt really address this comment: https://github.com/nvim-telescope/telescope.nvim/pull/2312#issuecomment-1479645530

i agree with tj here, that i dont really want builtin specific code in `pickers.lua`, because i dont think we should couple specific builtin code with the framework.

The framework should rather expose callbacks that allow us to do these things.

why can't we use `_on_input_filter_cb` its used in `_get_next_filtered_prompt` and we can then return the new prompt with `{ prompt = "asdf" }`

Reply:
I can look into that as well

Reply:
Hey @Conni2461 after a bit of investigation I realised why it would be a bad idea to put it into some callback — the biggest problem is the global state that we need to mutate and when the new picker is open we need to clear this state (or at least reset on reopening). I don't like an idea of stacking callbacks (we need to keep the functionality) and mutating global state inside of the stacked callbacks.

My proposal would be different. Define a new callback — `get_prompt_metadata` that will be executed in the picker and get the metadata from the prompt, then use this metadata to set the entry.lnum and entry.col on selection and also push this into the previewer.state. 

Thus we will never end up in a way when the global state is mutated. But every file-related picker will need to provide this `get_prompt_metadata` to achieve location parsing. 

If you are ok with this design please let me know — waiting for the green light.

Reply:
hmm, let me think about it. I am also not that happy with a new kinda specific callback (cant think of another reason why we would need that)

Reply:
I'll just put 3 things that are requirements for this feature (in my opinion): 

1. You need to mutate the prompt when the location is valid
2. You need to save the location and if it global state clear it for non supported picker
3. In my ideal world all of my pickers even external https://github.com/nvim-telescope/telescope-frecency.nvim will support this feature, or for instance I want to implement search by sapling files would support this feature, so having it as one flag option on the picker is preferred

Reply:
please be patient. Its been 3 days (, i reviewed it, i started looking into different implementations but i am also really busy. This project is my free time and i which is generally limited to weekends.

I'll merge a solution for this feature, but i need to think about this a couple more days. I'll not rush a solution because the majority of vim users tend to run the default branch, which makes breaking changes a pain.

I think you can run your fork a couple more days because there werent any mayor commits on master anyway.

Reply:
Sure, happy to understand that it is not lost.

Reply:
i think i have a implementation i slighly prefer: https://github.com/nvim-telescope/telescope.nvim/commit/64fc6c58f984b17ae243da90a75f3bafd23227df

i am not entirely happy and we need some util functions so people can easily set the callback but i think that makes it less coupled inside the picker.

so its not done, but i also havent spend more than 30minutes on it.

Its pretty likely that i dont have time to look at this prior to saturday, just a heads up

Reply:
Thank you for doing that, what confuses me in this is that the “filter_input_cb” sets totally global state which is super unclear side effect as for me. Just a thought, happy to help with any implementation you choose.

just let me know what I can do, thank you for your help on this!!

Reply:
yeah what i dont like about your implementation is that picker now does things that are totally only related to file specific things.

the picker itself, ui, sorting, etc, has nothing to do with files it operates on lists.

> telescope.nvim is a highly extendable fuzzy finder over lists. Built on the latest awesome features from neovim core. Telescope is centered around modularity, allowing for easy customization.

its also not a global global thing. i was thinking about introducing this local_state, which gets cleaned up after shutdown and allows users to save information, that is then accessible in the different stages of the picker (preview and actions). i am not happy that the default action/previewer have knowledge about this key/value pair, thats why i think its not done yet, but i think its a step up, because the picker no longer is responsible for it.